src/research/index.js

```diff
@@ -29,67 +29,65 @@ export const sources = [
   semantic,
   stardust,
   w3,
-]
+].map(source => ({
+  ...source,
+  components: source.components.map(component => ({
+    ...component,
+    source: source.name,
```

Comment:
Suggest either leaving ref to source object or naming key `sourceName` to follow suit with the rest.